src/sentry/tasks/store.py

```diff
@@ -43,10 +43,49 @@ def preprocess_event(cache_key=None, data=None, start_time=None, **kwargs):
         'project': project,
     })
 
+    # Iterate over all plugins looking for processors based on the input data
+    # plugins should yield a processor function only if it actually can operate
+    # on the input data, otherwise it should yield nothing
+    for plugin in plugins.all(version=2):
+        processors = plugin.get_event_preprocessors(data=data)
+        for processor in (processors or ()):
+            # On the first processor found, we just defer to the process_event
+            # queue to handle the actual work.
+            process_event.delay(cache_key=cache_key, start_time=start_time)
+            return
+
+    # If we get here, that means the event had no preprocessing needed to be done
+    # so we can jump directly to save_event
+    if cache_key:
+        data = None
+    save_event.delay(cache_key=cache_key, data=data, start_time=start_time)
+
+
+@instrumented_task(
+    name='sentry.tasks.store.process_event',
+    queue='events.process_event',
+    time_limit=65,
+    soft_time_limit=60,
+)
+def process_event(cache_key, start_time=None, **kwargs):
+    from sentry.plugins import plugins
+
+    data = default_cache.get(cache_key)
+
+    if data is None:
+        metrics.incr('events.failed', tags={'reason': 'cache', 'stage': 'pre'})
+        error_logger.error('preprocess.failed.empty', extra={'cache_key': cache_key})
+        return
+
+    project = data['project']
+    Raven.tags_context({
+        'project': project,
+    })
+
     # TODO(dcramer): ideally we would know if data changed by default
     has_changed = False
     for plugin in plugins.all(version=2):
-        processors = safe_execute(plugin.get_event_preprocessors, _with_transaction=False)
```

Comment:
I got rid of the safe_executes here since they don't provide any value, but I'm ok adding them back. If these raised an exception, tbh, I think we'd want to know about it personally. Not silently ignore here.